src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Helpers.cs

```diff
@@ -158,6 +159,80 @@ internal static void DisposeAll(this IEnumerable<IDisposable> disposables)
                 disposable.Dispose();
             }
         }
+
+        internal static bool IsSelfSigned(this X509Certificate2 cert)
+        {
+            X509SubjectKeyIdentifierExtension skid = null;
+            X509AuthorityKeyIdentifierExtension akid = null;
+
+            foreach (X509Extension extension in cert.Extensions)
+            {
+                if (extension.Oid.Value == Oids.SubjectKeyIdentifier)
+                {
+                    skid = (X509SubjectKeyIdentifierExtension)extension;
+                }
+                else if (extension.Oid.Value == Oids.AuthorityKeyIdentifier)
+                {
+                    akid = new X509AuthorityKeyIdentifierExtension();
+                    akid.CopyFrom(extension);
+                }
+
+                if (akid != null && skid != null)
+                {
+                    break;
+                }
+            }
+
+            return IsSelfSigned(cert, skid, akid);
+        }
+
+        internal static bool IsSelfSigned(
+            this X509Certificate2 cert,
+            X509SubjectKeyIdentifierExtension skid,
+            X509AuthorityKeyIdentifierExtension akid)
+        {
+            Debug.Assert(cert != null);
+
+            if (!cert.SubjectName.RawData.ContentsEqual(cert.IssuerName.RawData))
```

Comment:
https://tools.ietf.org/html/rfc3280#section-9
> The X.500 series of specifications defines rules for
comparing distinguished names that require comparison of strings
without regard to case, character set, multi-character white space
substring, or leading and trailing white space.  This specification
relaxes these requirements, requiring support for binary comparison
at a minimum.

> CAs MUST encode the distinguished name in the subject field of a CA
certificate identically to the distinguished name in the issuer field
in certificates issued by that CA.  If CAs use different encodings,
implementations might fail to recognize name chains for paths that
include this certificate.  As a consequence, valid paths could be
rejected.

https://tools.ietf.org/html/rfc5280#section-7.1
> Two distinguished names DN1 and DN2 match if they
have the same number of RDNs, for each RDN in DN1 there is a matching
RDN in DN2, and the matching RDNs appear in the same order in both
DNs.

ITU T-REC X.509 says it follows the `distinguishedNameMatch` rule, which does a long-winded way of saying `binary match`, since it's dealing with it at concepts instead of encoded values.